pkg/scan/docker/docker.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2021 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/daemon"
+	"github.com/mholt/archiver/v3"
+	"github.com/palantir/log4j-sniffer/pkg/crawl"
+	"github.com/palantir/log4j-sniffer/pkg/scan"
+	"github.com/pkg/errors"
+)
+
+func ScanImages(ctx context.Context, config scan.Config, stdout, stderr io.Writer, apiClient client.CommonAPIClient, imageExtractionDir string) (int64, error) {
+	scanner := scan.NewScannerFromConfig(config, stdout, stderr)
```

Comment:
Per comment in command, I think it would make more sense to have `scanner` as an argument to `ScanImages` (rather than taking the config and then instantiating scanner here)